macOS/DuckDuckGo/Bookmarks/Model/BookmarkManager.swift

```diff
@@ -203,10 +203,10 @@ final class LocalBookmarkManager: BookmarkManager {
     }
 
     @discardableResult func makeBookmark(for url: URL, title: String, isFavorite: Bool) -> Bookmark? {
-        makeBookmark(for: url, title: title, isFavorite: isFavorite, index: nil, parent: nil)
+        makeBookmark(for: url, title: title, isFavorite: isFavorite, index: nil, parent: nil, completion: { _ in })
     }
 
-    @discardableResult func makeBookmark(for url: URL, title: String, isFavorite: Bool, index: Int? = nil, parent: BookmarkFolder? = nil) -> Bookmark? {
+    @discardableResult func makeBookmark(for url: URL, title: String, isFavorite: Bool, index: Int? = nil, parent: BookmarkFolder? = nil, completion: @escaping (Error?) -> Void) -> Bookmark? {
```

Comment:
I know that the idea of this is to make tests more reliable, but the completion is confusing. We do not call it when it is complete; we only call it when an error is saved to the database.

I would clarify that it is for error only by making the Error non-nullable.

Reply:
We’re calling the completion after save is complete, regardless of whether the error happened. The error parameter is optional, but the completion handler is called in all cases.

See the place where save happens (LocalBookmarkStore) and completion is called both on didSave and on error:
<img width="1177" alt="Screenshot 2025-03-19 at 15 31 22" src="https://github.com/user-attachments/assets/222a09d6-c046-42e5-9925-9f5c4e8c5f73" />

Reply:
After syncing on MM I was made aware that I forgot to add the completion handler for the error case. The fix is pushed now (we’re interested in calling the completion handler in all cases of course) 💪